portal/src/OnboardingSurveyScreen.tsx

```diff
@@ -697,48 +714,69 @@ function Step4(_props: StepProps) {
   const prefix = "step4";
   const reasonChoiceGroup = "reasonChoiceGroup";
   const reasonChoices = ["Auth", "SSO", "Security", "Portal", "Other"];
-  const reasonChoicesFromLocalStorage = getFromLocalStorage("reasonChoices");
+  const reasonChoicesFromLocalStorage = getFromLocalStorage("use_cases") as
+    | UseCase[]
+    | undefined;
   const defaultReasonChoices: string[] = (
     reasonChoicesFromLocalStorage === undefined
       ? []
-      : reasonChoicesFromLocalStorage
+      : reasonChoicesFromLocalStorage.filter((elem) => typeof elem === "string")
   ) as string[];
   const [reasonChoicesState, setReasonChoicesState] =
     useState(defaultReasonChoices);
   const empty = useMemo(() => {
     return reasonChoicesState.length === 0;
   }, [reasonChoicesState]);
   const { renderToString } = useContext(Context);
-  const otherReasonFromLocalStorage = getFromLocalStorage("otherReason");
-  const [otherReason, setOtherReason] = useState(
-    otherReasonFromLocalStorage === undefined
-      ? ""
-      : (otherReasonFromLocalStorage as string)
-  );
+  let otherReasonFromLocalStorage = "";
+  if (reasonChoicesFromLocalStorage !== undefined)
+    reasonChoicesFromLocalStorage.forEach((elem) => {
+      if (typeof elem !== "string")
+        otherReasonFromLocalStorage = elem.other_reason;
+    });
+  const [otherReason, setOtherReason] = useState(otherReasonFromLocalStorage);
   const navigate = useNavigate();
+  const capture = useCapture();
   useEffect(() => goToFirstUnfilled(4, navigate), [navigate]);
   const onClickNext = useCallback(
     (e) => {
       e.preventDefault();
       e.stopPropagation();
-      setLocalStorage("reasonChoices", reasonChoicesState);
-      if (otherReason !== "") setLocalStorage("otherReason", otherReason);
+      const constructUseCases = reasonChoicesState as UseCase[];
+      if (constructUseCases.includes("Other")) {
+        const i = constructUseCases.indexOf("Other", 0);
```

Comment:
Use a enum to represent `"Other"` instead of copying the string literal many times.

```
enum ReasonChoices {
  Auth = "Auth",
  Other = "Other",
  // ...remaining 
}
```